test/shared/HttpParsingData.cs

```diff
@@ -358,79 +358,79 @@ public static IEnumerable<string> TargetWithNullCharData
         public static IEnumerable<object[]> RequestHeaderInvalidData => new[]
         {
             // Missing CR
-            new[] { "Header: value\n\r\n", CoreStrings.FormatBadRequest_InvalidRequestHeader_Detail(@"Header: value\x0A") },
-            new[] { "Header-1: value1\nHeader-2: value2\r\n\r\n", CoreStrings.FormatBadRequest_InvalidRequestHeader_Detail(@"Header-1: value1\x0A") },
-            new[] { "Header-1: value1\r\nHeader-2: value2\n\r\n", CoreStrings.FormatBadRequest_InvalidRequestHeader_Detail(@"Header-2: value2\x0A") },
+            new[] { "Header: value\n\r\n", CoreStrings.FormatBadRequest_InvalidRequestHeader_Detail(@"Header: value") },
+            new[] { "Header-1: value1\nHeader-2: value2\r\n\r\n", CoreStrings.FormatBadRequest_InvalidRequestHeader_Detail(@"Header-1: value1") },
+            new[] { "Header-1: value1\r\nHeader-2: value2\n\r\n", CoreStrings.FormatBadRequest_InvalidRequestHeader_Detail(@"Header-2: value2") },
```

Comment:
https://github.com/aspnet/KestrelHttpServer/pull/3068#pullrequestreview-170870112

I think we should change the error message whenever we find a bad CR/LF and then it would be pretty easy to understand. Something like `Unexpected CR or LF in header: '{detail}'`.

Reply:
Could even put a separate message up for the two different cases:

`Unexpected CR in header: '{detail}'` or possibly `Unexpected CR without LF in header: '{detail}'` (although this sort of infers that the header is fine other than that, which isn't necessarily true).

Now that I'm looking at it, I could also rewind the reader and take the bad CR or LF into the detail message... Hmm. Maybe that is the right place to start... 

`Unexpected LF in header: '{detail}'`

Reply:
I like distinct error messages. Even just `Unexpected (CR|LF) in header: '{header up until the CR or LF}'` any time you see one without the other is enough for me.